src/lib.rs

```diff
@@ -367,33 +347,33 @@ impl SecretKey {
     /// has been copied onto the heap.
     pub fn from_mut(fr: &mut Fr) -> Self {
         let fr_ptr = fr as *mut Fr;
-        let mut boxed_fr = Box::new(Fr::zero());
+        let mut res_fr = Fr::zero();
         unsafe {
-            copy_nonoverlapping(fr_ptr, &mut *boxed_fr as *mut Fr, 1);
+            copy_nonoverlapping(fr_ptr, &mut res_fr as *mut Fr, 1);
         }
         clear_fr(fr);
-        SecretKey(boxed_fr)
+        SecretKey(res_fr)
```

Comment:
Now I'm confused why we can't just copy `fr`. That should have the same effect, shouldn't it?
```rust
let sk = SecretKey(*fr);
clear_fr(fr);
sk
```
Do we still need this function at all?

Reply:
Good catch